src/niraj/narcissisticNumbers.ts

```diff
@@ -0,0 +1,40 @@
+/**
+ * Function to check whether a given number is a narcissistic number or not
+ * @param {number} numberToCheck - The number to check
+ * @return {boolean}
+ */
+
+export const isNarcissisticNumber = (numberToCheck: number): boolean => {
+  if (numberToCheck <= 0) return false;
+
+  // All numbers between 1-10 are narcissistic number
+  if (numberToCheck > 0 && numberToCheck < 10) return true;
+
+  const strNum = String(numberToCheck);
```

Comment:
We can define the type of 'strNum'.

Reply:
Please help to clarify my doubt. Do we need to do type check for every intermediate value?
- If we are using any js functions which always outputs a particular type (String -> string, Math.sqrt -> number), it already knows the type right?
- For the below code, even though type is not declared for `varB` and `varA`, at the time of using it editor is still showing the type as number as it reflected from the return type of the function? Do we still need to define types for immediate variables?
```
const myFunction = (): number => 4;
const varB = myFunction();
const varA = varB + 3;
console.log(varA, varB);
```

Reply:
You are right Raj. Wherever typescript can automatically infer a variable's type, you do not have to provide its type.